src/coreclr/gc/gc.cpp

```diff
@@ -17976,6 +17944,63 @@ size_t gc_heap::current_generation_size (int gen_number)
     return gen_size;
 }
 
+#ifdef USE_REGIONS
+// We may need a new empty region while doing a GC so try to get one now, if we don't have any
+// reserve in the free region list. 
+bool gc_heap::try_get_new_free_region()
+{
+    heap_segment* region = 0;
+    if (free_regions)
+    {
+        dprintf (REGIONS_LOG, ("h%d has %d free regions %Ix", heap_number, num_free_regions,
+            heap_segment_mem (free_regions)));
+        return true;
+    }
+    else
+    {
+        region = allocate_new_region (__this, 0, false);
+        if (region)
+        {
+            init_table_for_region (0, region);
```

Comment:
How about the case where init_table_for_region fails because the mark array cannot be committed?  It seems that we would return true in this case, but likely still fail when we later try to get the region.

Reply:
good catch! I did check for this (I set region to 0 in `init_table_for_region` when we can't commit) but I made a mistake when I refactored it out to its own method. should have `init_table_for_region` return false if we can't commit.